### ISSUE TITLE
Slight tutorial 

### ISSUE THREAD

**Issue by rockerBOO** [aosasona/migrant#1]
Just writing this down as it was unclear how to create the migrations. Feel free to use any part of this in the documentation or I can make a PR if desired. Thanks for making this!

---

Migrations files use up and down and should be prefixed with a number to keep it in order. Create both up and down for each migrations.

Match the directory pattern with how you have it in the code. 

```
priv/migrations/00001_create_users.up.sql
priv/migrations/00001_create_users.down.sql
```

Spec:

```
<name>.<up/down>.sql e.g 00001_create_users.up.sql
```

Reference code: 

```gleam
import gleam/erlang
import app/database
import migrant

pub fn main() {
 let db = database.connect()

 // app is your application name in gleam
 let assert Ok(priv_directory) = erlang.priv_directory("app")
 let assert Ok(_) = migrant.migrate(db, priv_directory <> "/migrations")

 Nil
}
```

**Comment by aosasona** [comment]
Thanks for this! Sorry it took you time to figure it out, I worked on this in a hurry and I haven't really had time to look back on it since, I should rewrite the whole thing properly in the future and document everything, but for now, I will pin this for anyone who needs it.

**Comment by aosasona** [comment]
Here's also a reference:

https://github.com/aosasona/jsorm/tree/master/priv/migrations

I didn't expect anyone to really use it but I had to publish it to make it easier to deploy my other projects that used it.

**Comment by rockerBOO** [comment]
> Thanks for this! Sorry it took you time to figure it out, I worked on this in a hurry and I haven't really had time to look back on it since, I should rewrite the whole thing properly in the future and document everything, but for now, I will pin this for anyone who needs it.

No worries really. Didn't take much time to figure out just didn't know how to do it. Works well enough for what I need but no need to worry about making improvements. If I find any improvements, I will try to get them to you. 

Only thing right now would be how to roll back a migration but not a big need right now. I was doing raw SQL and manually updating the db so this works fine for now. 

Again thanks for making this. Helps a lot with being able to invest in making changes more fluidly and be able to deploy to production.

**Comment by aosasona** [comment]
> > Thanks for this! Sorry it took you time to figure it out, I worked on this in a hurry and I haven't really had time to look back on it since, I should rewrite the whole thing properly in the future and document everything, but for now, I will pin this for anyone who needs it.
> 
> 
> 
> No worries really. Didn't take much time to figure out just didn't know how to do it. Works well enough for what I need but no need to worry about making improvements. If I find any improvements, I will try to get them to you. 
> 
> 
> 
> Only thing right now would be how to roll back a migration but not a big need right now. I was doing raw SQL and manually updating the db so this works fine for now. 
> 
> 
> 
> Again thanks for making this. Helps a lot with being able to invest in making changes more fluidly and be able to deploy to production. 

Yeah, I wanted to be able to have my code and migrations self-contained for deploying to fly (and obviously didn't want duplicates; ALTERs can be brutal haha), so I quickly threw this together but yes, I'll look at making it more trustworthy and maybe a CLI (CLI libraries in Gleam seem to be picking up) and API for rollbacks, might be a while since I haven't needed it yet. 

Unrelated but I love the awesome Neovim list, used it a while back, I had no idea that was you!
